cpp/FileTransferAgent.cpp

```diff
@@ -283,19 +293,31 @@ void Snowflake::Client::FileTransferAgent::uploadFilesInParallel(std::string *co
       CXX_LOG_DEBUG("Sequential upload %d th file %s start", i, metadata->srcFileName.c_str());
       do
       {
-        RemoteStorageRequestOutcome outcome = uploadSingleFile(m_storageClient, metadata,
-          resultIndex);
+        RemoteStorageRequestOutcome outcome = RemoteStorageRequestOutcome::SUCCESS;
+        if(isPutFastFailEnabled() && !failedTransfers.empty())
+        {
+          m_executionResults->SetTransferOutCome(RemoteStorageRequestOutcome::SKIP_UPLOAD_FILE, resultIndex);
+          CXX_LOG_DEBUG("Sequential upload, put fast fail enabled, Skipping file.");
+          break;
+        }
```

Comment:
When a put fails, we append to the string the filename that failed to upload. 
If fast fail is enabled and there already failed a file to upload then we skip the upload of the rest of the files.